### PR TITLE
C3p0 Replacement

### PR DESCRIPTION
**Pull request by drecchia**

### CODE REVIEW

File: .checkstyle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tananaev) [inline .checkstyle]: Why do we need this file? What does it do?

File: src/org/traccar/database/DataManager.java [hunk 2]
```diff
@@ -106,19 +110,23 @@ private void initDatabase() throws Exception {
...
-            ds.setTestConnectionOnCheckin(true);
-            ds.setMaxStatementsPerConnection(config.getInteger("database.maxStatements"));
+            HikariConfig hConfig = new HikariConfig();
+            hConfig.setDriverClassName(config.getString("database.driver"));
+            hConfig.setJdbcUrl(config.getString("database.url"));
+            hConfig.setUsername(config.getString("database.user"));
+            hConfig.setPassword(config.getString("database.password"));
+            hConfig.setConnectionInitSql("SELECT 1 FROM DUAL");
+            hConfig.setIdleTimeout(600000); // milliseconds
+//            ds.setIdleConnectionTestPeriod(600);
```
Comment (tananaev) [inline src/org/traccar/database/DataManager.java]: Please remove comments.

File: src/org/traccar/database/DataManager.java [hunk 3]
```diff
@@ -106,19 +110,23 @@ private void initDatabase() throws Exception {
...
 
-            ComboPooledDataSource ds = new ComboPooledDataSource();
-            ds.setDriverClass(config.getString("database.driver"));
-            ds.setJdbcUrl(config.getString("database.url"));
-            ds.setUser(config.getString("database.user"));
-            ds.setPassword(config.getString("database.password"));
-            ds.setIdleConnectionTestPeriod(600);
-            ds.setTestConnectionOnCheckin(true);
-            ds.setMaxStatementsPerConnection(config.getInteger("database.maxStatements"));
+            HikariConfig hConfig = new HikariConfig();
```
Comment (tananaev) [inline src/org/traccar/database/DataManager.java]: Can you please rename it to `poolConfig` or `hikariConfig` or something like that.

**Comment by tananaev** [comment]
Mostly looks good. Please fix minor issues and merge conflicts.

**Comment by tananaev** [comment]
Why have you closed the PR?

Note that I don't get a notification about commits, so when you fix something, you need to put a comment here.

**Comment by drecchia** [comment]
Sorry Anton, I have created another pull request.

**Pull request closed without merging**